src/main/clojure/clara/tools/tracing.cljc

```diff
@@ -114,3 +114,48 @@
                          (first))]
     (.-trace ^PersistentTracingListener listener)
     (throw (ex-info "No tracing listener attached to session." {:session session}))))
+
+(defn ^:private node-id->productions
+  "Given a session and a node ID return a list of the rule and query names associated
+   with the node."
+  [session id]
+  (let [node (-> session .-rulebase :id-to-node (get id))]
+    (into []
+          (comp
+           (map second)
+           cat
+           (map second))
+          (eng/get-conditions-and-rule-names node))))
+
+(defn traced-session->ranked-productions
```

Comment:
I think I'd probably prefer the name to just be`ranked-productions`.

I'm not sure there is a reason to express "conversion" that the `x->y` tends to imply.
I don't 100% agree [1] with the suggestions of this post [2], but do think it has some fairly good guidance in this regard

[1] It does mention 
> There are exceptions to every rule.
So it covers that base.

[2] https://stuartsierra.com/2016/01/09/how-to-name-clojure-functions

Reply:
That name is OK with me, will change.